DQMOffline/Trigger/plugins/RazorMonitor.cc

```diff
@@ -0,0 +1,421 @@
+}
+void RazorMonitor::bookME(DQMStore::IBooker &ibooker, RazorME& me, const std::string& histname, const std::string& histtitle, int nbinsX, double xmin, double xmax, int nbinsY, double ymin, double ymax)
+{
+  me.numerator   = ibooker.book2D(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, xmin, xmax, nbinsY, ymin, ymax);
+  me.denominator = ibooker.book2D(histname+"_denominator", histtitle+" (denominator)", nbinsX, xmin, xmax, nbinsY, ymin, ymax);
+}
+void RazorMonitor::bookME(DQMStore::IBooker &ibooker, RazorME& me, const std::string& histname, const std::string& histtitle, const std::vector<double>& binningX, const std::vector<double>& binningY)
+{
+  int nbinsX = binningX.size()-1;
+  std::vector<float> fbinningX(binningX.begin(),binningX.end());
+  float* arrX = &fbinningX[0];
+  int nbinsY = binningY.size()-1;
+  std::vector<float> fbinningY(binningY.begin(),binningY.end());
+  float* arrY = &fbinningY[0];
+
+  me.numerator   = ibooker.book2D(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, arrX, nbinsY, arrY);
+  me.denominator = ibooker.book2D(histname+"_denominator", histtitle+" (denominator)", nbinsX, arrX, nbinsY, arrY);
+}
+
+void RazorMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+				 edm::Run const        & iRun,
+				 edm::EventSetup const & iSetup) 
+{  
+  
+  std::string histname, histtitle;
+
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+
+  // 1D hist, MR
+  histname = "MR"; histtitle = "PF MR";
+  bookME(ibooker,MR_ME_,histname,histtitle,mr_binning_);
+  setMETitle(MR_ME_,"PF M_{R} [GeV]","events / [GeV]");
+
+  // profile, MR vs LS
+  histname = "MRVsLS"; histtitle = "PF MR vs LS";
+  bookME(ibooker,MRVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax, mr_binning_.front(),mr_binning_.back());
+  setMETitle(MRVsLS_,"LS","PF M_{R} [GeV]");
+
+  // 1D hist, MR, with tight jet requirement
+  histname = "MR_Tight"; histtitle = "PF MR";
+  bookME(ibooker,MR_Tight_ME_,histname,histtitle,mr_binning_);
+  setMETitle(MR_Tight_ME_,"PF M_{R} [GeV]","events / [GeV]");
+
+  // 1D hist, Rsq
+  histname = "Rsq"; histtitle = "PF Rsq";
+  bookME(ibooker,Rsq_ME_,histname,histtitle,rsq_binning_);
+  setMETitle(Rsq_ME_,"PF R^{2}","events");
+
+  // profile, Rsq vs LS
+  histname = "RsqVsLS"; histtitle = "PF Rsq vs LS";
+  bookME(ibooker,RsqVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax, rsq_binning_.front(), rsq_binning_.back());
+  setMETitle(RsqVsLS_,"LS","PF R^{2}");
+
+  // 1D hist, Rsq, with tight jet requirement
+  histname = "Rsq_Tight"; histtitle = "PF Rsq";
+  bookME(ibooker,Rsq_Tight_ME_,histname,histtitle,rsq_binning_);
+  setMETitle(Rsq_Tight_ME_,"PF R^{2}","events");
+
+  // 1D hist, dPhiR
+  histname = "dPhiR"; histtitle = "dPhiR";
+  bookME(ibooker,dPhiR_ME_,histname,histtitle,dphiR_binning_);
+  setMETitle(dPhiR_ME_,"dPhi_{R}","events");
+
+  // profile, dPhiR vs LS
```

Comment:
are you expecting to have a different efficiency value in dPhiR w.r.t. Rsq or MR ?
I would expect the same value which is driven by your online and offline selection
=> I do not get why you need 3 plots vs LS done in this way ....